src/Raven.Server/Documents/Transformers/TransformationScope.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Raven.Server.Documents.Indexes.Persistence.Lucene.Documents;
+using Raven.Server.Documents.Indexes.Static;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Transformers
+{
+    public class TransformationScope : IDisposable
+    {
+        private readonly DocumentsOperationContext _context;
+
+        public TransformationScope(IndexingFunc transformResults, DocumentDatabase documentDatabase, DocumentsOperationContext context)
+        {
+            _context = context;
+            CurrentTransformationScope.Current = new CurrentTransformationScope(transformResults, documentDatabase);
+        }
+
+        public void Dispose()
+        {
+            CurrentTransformationScope.Current = null;
+        }
+
+        public IEnumerable<Document> Transform(IEnumerable<Document> documents)
+        {
+            var docsEnumerator = new StaticIndexDocsEnumerator(documents, CurrentTransformationScope.Current.TransformResults, null, StaticIndexDocsEnumerator.EnumerationType.Transformer);
+
+            IEnumerable transformedResults;
+            while (docsEnumerator.MoveNext(out transformedResults))
+            {
+                using (docsEnumerator.Current.Data)
+                {
+                    var values = new DynamicJsonArray();
+                    var result = new DynamicJsonValue
+                    {
+                        ["$values"] = values
+                    };
+
+                    PropertyAccessor accessor = null;
+                    foreach (var transformedResult in transformedResults)
```

Comment:
There are two enumerators here. First one 'docsEnumerator' enumerates through the documents and the second one (foreach) is enumerating through transformation function results. I'm returning one document with array of values from the transformation. It shouldn't be a problem. Since I've only done some basic transformations on Friday I cannot confirm that, but tomorrow I will write a test to check it.